databricks/koalas/frame.py

```diff
@@ -8529,21 +8529,41 @@ def query(self, expr, inplace=False):
             * Internal columns that starting with __. are able to access,
               however, they are not supposed to be accessed.
             * This delegates to Spark SQL so the syntax follows Spark SQL
-            * If you want the exactly same syntax with pandas' you can work around
-              by using :meth:`DataFrame.map_in_pandas`. See the example below.
+            * If you want the exactly same syntax with pandas you can work around
+              by using :meth:`DataFrame.map_in_pandas`, when only the size of the
+              data is smaller than `compute.shortcut_limit`. See the example below.
 
+                >>> from databricks.koalas.config import set_option, reset_option, get_option
                 >>> df = ks.DataFrame([(1, 2), (3, 4), (5, 6)], columns=['A', 'B'])
                 >>> df
                    A  B
                 0  1  2
                 1  3  4
                 2  5  6
-                >>> num = 1
+                >>> # You can check the value of `compute.shortcut_limit` like the below.
+                ... get_option('compute.shortcut_limit')
```

Comment:
Koalas targets big data. I think there is not so much point to show how to work around with small dataset.

Reply:
i have some tested, the index will be lost after map_in_pandas (data looks okay)

```python
>>> df = ks.DataFrame({'A': range(1000), 'B': range(1000)})
>>> def query_func(pdf) -> ks.DataFrame[int, int]:
...     num = 990
...     return pdf.query('A > @num')
...
>>> df.map_in_pandas(query_func)
    c0   c1
0  991  991
1  992  992
2  993  993
3  994  994
4  995  995
5  996  996
6  997  997
7  998  998
8  999  999
```

is it expected behaviour? or just a bug?

Reply:
Yea, that's expected at this moment. See https://github.com/databricks/koalas/issues/1307